### PR TITLE
FOGL-2585 - Do not allow deletion of Notification service if notifications exist

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: python/foglamp/services/core/api/service.py [hunk 1]
```diff
@@ -137,7 +145,7 @@ def get_service_records():
 
         if name is None:
             raise web.HTTPBadRequest(reason='Missing name property in payload.')
-        if utils.check_reserved(name) is False:
+        if utils.check_reserved(name) is False or utils.check_foglamp_reserved(name) is False:
             raise web.HTTPBadRequest(reason='Invalid name property in payload.')
```
Comment (praveen-garg) [inline python/foglamp/services/core/api/service.py]: We should make this message better.

File: python/foglamp/common/utils.py [hunk 2]
```diff
@@ -37,6 +37,27 @@ def check_reserved(string):
...
+        'foglamp',
+        'general',
+        'advanced',
+        'notifications',
+        'north',
+        'south',
+        'filter',
+        'notify',
+        'rule',
+        'delivery',
```
Comment (praveen-garg) [inline python/foglamp/common/utils.py]: rule and delivery are also reserved?

Comment (amarendra-dianomic) [reply python/foglamp/common/utils.py]: yes, these have a pre defined meaning in Foglamp, so should not be used as service name.

File: python/foglamp/services/core/api/service.py (hunk 1 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/api/service.py]: done

**Comment by amarendra-dianomic** [comment]
@praveen-garg all done, pl check.

**Pull request merged**